### PR TITLE
[Doc] How to: Enable Overriding Subscription ID in Your Module

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

Add a development document "How to: Enable Overriding Subscription ID in Your Module".
Code in #15659

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: documentation/development-docs/enable-subscription-id-overriding.md [hunk 1]
```diff
@@ -1,12 +1,104 @@
...
+1. `GetDynamicParameters()` is decorated with [`new` modifier](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/new-modifier).
+1. Call `base.GetDynamicParameters()` and combine the results in your `GetDynamicParameters()`.
+
+A sample implementation:
+
+```csharp
+public new object GetDynamicParameters()
+{
+    var parameters = base.GetDynamicParameters() as RuntimeDefinedParameterDictionary;
+    AzureRmProfile localProfile = DefaultProfile as AzureRmProfile;
```
Comment (BethanyZhou) [inline documentation/development-docs/enable-subscription-id-overriding.md:65]: Do these mean a cmdlet should add Line64 and then implement its own features like Line 65-76 if it has implemented `IDynamicParameters`

Comment (isra-fel) [reply documentation/development-docs/enable-subscription-id-overriding.md:65]: Yes you are right, I should probably make 65-76 more abstract so it won't attract too much attention

Comment (isra-fel) [reply documentation/development-docs/enable-subscription-id-overriding.md]: Hi @BethanyZhou I rewrote the code sample. Please review.  Thanks!

**Comment by BethanyZhou** [comment]
LGTM

**Review by BethanyZhou — APPROVED** [review]

**Pull request merged**